pkg/controller/jobframework/setup.go

```diff
@@ -110,3 +157,88 @@ func SetupIndexes(ctx context.Context, indexer client.FieldIndexer, opts ...Opti
 		return nil
 	})
 }
+
+func isAPIAvailable(mgr ctrl.Manager, rcApiNames []string, toApiNames []string) bool {
+	discoveryClient := discovery.NewDiscoveryClientForConfigOrDie(mgr.GetConfig())
+	apiGroupResources, err := restmapper.GetAPIGroupResources(discoveryClient)
```

Comment:
`GetAPIGroupResources` [returns](https://github.com/kubernetes/client-go/blob/592d891671b2a09e5f81781b28ebe078d8115e41/restmapper/discovery.go#L147) a pointer to the slice. Better to check for nil (in case it's a fresh cluster), or does `NewDiscoveryRESTMapper` take care of it?